### DISCUSSION TITLE
Adopt NX to the project

### DISCUSSION THREAD

**Discussion by pavankjadda** [stevermeister/ngx-cookie-service#240] [Ideas]
We now have 2 projects based on Angular CLI

- ngx-cookie-service
- ngx-cookie-service-ssr

The project structure is based on Angular CLI. [NX](https://nx.dev/) has gained popularity in recent years and I found it more useful for library authors. Here are pros/cons of NX
Pros:

- Testing is way easier. If you want to test local changes, we need to use `npm link` which is not reliable and buggy. NX makes so easier to do this
- Mono repo support
- Publish examples in same repo as separate project

Cons:

- We need to wait until NX supports after new Angular version releases. Usually with in a week

**Comment by pavankjadda** [comment]
@stevermeister any thoughts on this

**Comment by stevermeister** [comment]
I think NX is cool. 
it's OK, to wait until the new version of Angular, that's what you are paying for extra features. 

a more critical downside that I see - is extra complexity for other contributors. 

but still... LET'S DO IT!
